modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/UserAccountResourceImpl.java

```diff
@@ -1009,13 +979,20 @@ else if (StringUtil.equalsIgnoreCase(
 
 		ServiceContext serviceContext = _createServiceContext(userAccount);
 
-		user = _userService.updateExternalReferenceCode(
+		_userService.updateExternalReferenceCode(
 			userAccountId,
```

Comment:
this "user = " removal didn't make sense to me, but SF was complaining about it:

> No need to assign call to variable 'user', because 'user' is reassigned another value without being used

Reply:
probably if we move the block below (see the other comment) on top we can re-add this assignment

Reply:
or here we can try replacing it with
```
user = _userService.updateStatus(
			userAccountId, workflowStatus, serviceContext);
```

```
user = _userService.updateStatus(
			user.getUserId, workflowStatus, serviceContext);
```

I think it should be the same but because we are using it (user.getUserId) it should not complain. But please try it to verify that it works fine

Reply:
The other changes makes sense. But i think this will need to be removed either way because of [this](https://github.com/liferay/liferay-portal/blob/master/modules/util/source-formatter/src/main/resources/documentation/check/unnecessary_assign_check.markdown)

Reply:
The ERC is being updated and right after the Status is being updated, so SF thinks the reassignment is not needed since there's nothing changing between the ERC update and the Status update